Sources/MapboxMaps/Location/LocationProducer.swift

```diff
@@ -112,17 +105,32 @@ internal final class LocationProducer: LocationProducerProtocol {
         }
     }
 
+    private let interfaceOrientationProvider: InterfaceOrientationProvider
+    private let notificationCenter: NotificationCenterProtocol
+    private let device: UIDevice
     private var mayRequestWhenInUseAuthorization: Bool
+    private var headingOrientationUpdateBackupTimer: Timer?
+    private weak var userInterfaceOrientationView: UIView?
 
     internal init(locationProvider: LocationProvider,
+                  interfaceOrientationProvider: InterfaceOrientationProvider,
+                  notificationCenter: NotificationCenterProtocol,
+                  userInterfaceOrientationView: UIView,
+                  device: UIDevice,
                   mayRequestWhenInUseAuthorization: Bool) {
         self.locationProvider = locationProvider
+        self.notificationCenter = notificationCenter
         self.mayRequestWhenInUseAuthorization = mayRequestWhenInUseAuthorization
         self.latestAccuracyAuthorization = locationProvider.accuracyAuthorization
+        self.interfaceOrientationProvider = interfaceOrientationProvider
+        self.userInterfaceOrientationView = userInterfaceOrientationView
+        self.device = device
         self.locationProvider.setDelegate(self)
     }
 
     deinit {
+        headingOrientationUpdateBackupTimer?.invalidate()
```

Comment:
should we also tell the device to `endGeneratingDeviceOrientationNotifications` on deinit if it's currently generating?